coriolisclient/cli/replica_schedules.py

```diff
@@ -0,0 +1,135 @@
+# Copyright (c) 2017 Cloudbase Solutions Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Command-line interface sub-commands related to replicas.
+"""
+
+from cliff import command
+from cliff import lister
+from cliff import show
+
+from coriolisclient.cli import formatter
+
+
+class ReplicaScheduleFormatter(formatter.EntityFormatter):
+
+    columns = ("Replica ID",
+               "ID",
+               "Schedule",
+               "Created",
+               )
+
+    def _get_sorted_list(self, obj_list):
+        return sorted(obj_list, key=lambda o: o.created_at)
+
+    def _get_formatted_data(self, obj):
+        data = (obj.replica_id,
+                obj.id,
+                obj.schedule,
+                obj.created_at,
+                )
+        return data
+
+
+class ReplicaScheduleDetailFormatter(formatter.EntityFormatter):
+
+    columns = ("id",
+               "replica_id",
+               "schedule",
+               "created",
+               "last_updated",
+               "enabled",
+               "expires",
+               )
+
+    def _get_formatted_data(self, obj):
+        data = (obj.id,
+                obj.replica_id,
+                obj.schedule,
+                obj.created_at,
+                obj.updated_at,
+                obj.enabled,
+                obj.expiration_date,
+                )
+        return data
+
+
+class CreateReplicaSchedule(show.ShowOne):
+    """Start a replica schedule"""
+    def get_parser(self, prog_name):
+        parser = super(CreateReplicaSchedule, self).get_parser(prog_name)
+        parser.add_argument('--replica',
+                            help='The ID of the replica')
+        parser.add_argument('--schedule',
```

Comment:
Should we pass a cron string to make it easier like "* 1 * * *" for the users?

Reply:
I was thinking of adding a cron string parser :)